bin/tictactoe

```diff
@@ -1,3 +1,41 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+puts "Hello! Welcome to Tic Tac Toe! Let's play :)"
+input = nil
+while input != "no" || input != "No" || input != "NO"
+  puts "How many players are there? 0, 1, or 2?"
+  input = gets.chomp
+  if input == "1"
+    game = Game.new(Human.new("X"), Player::Computer.new("O"))
+    until game.won?
+      game.play
+    end
+    puts "Great game! Want to play again? Yes or No?"
+    input = gets.chomp
+    if input == "no" || input == "No" || input == "NO"
+      puts "Thanks for playing!"
+    end
+  elsif input == "2"
+    game = Game.new(Human.new("X"), Human.new("O"))
+    until game.won?
+      game.play
+    end
+    puts "Great game! Want to play again? Yes or No?"
+    input = gets.chomp
+      if input == "no" || input == "No" || input == "NO"
+        puts "Thanks for playing!"
+      end
+  elsif input == "0"
+    game = Game.new(Player::Computer.new("X"), Player::Computer.new("O"))
+    until game.won?
+      game.play
+    end
+    puts "Great game! Want to play again? Yes or No?"
+    input = gets.chomp
+      if input == "no" || input == "No" || input == "NO"
+        puts "Thanks for playing!"
+      end
+    end
+end
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.